tests/services/UserValidationService.spec.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable @typescript-eslint/unbound-method */
```

Comment:
Similar to other points, update indentation. Sometimes it's unclear what belongs to what. Just two points

Point 1: 
remember that you are using objects in here. In the last couple of tests, you are updating the `password` property of an object. I'm not sure about mocha, but Karma in Angular runs tests in random order and I'd expect the same of mocha. You could change password of an object and have false-pass or false-deny (you would be using a different password than you are testing). If mocha does run them in random order, it is best to ALWAYS create a new object. Karma has `beforeEach` and I would expect something similar to exist in mocha.

Point 2:
Love the amount of tests and their cases. Great job!